tests/testthat/test-step-join.R

```diff
@@ -80,7 +80,7 @@ test_that("simple left joins use [", {
 
   expect_equal(
     dt1 %>% left_join(dt2, by = "x") %>% show_query(),
-    expr(dt2[dt1, on = .(x), allow.cartesian = TRUE])
+    call2("setcolorder", expr(dt2[dt1, on = .(x), allow.cartesian = TRUE]), c("x", "a", "b"))
```

Comment:
Why are you using `call2()` here?

Reply:
I first had the simpler form without `call2()` but this then the argument `c("x", "a", "b")` differs (language vs character vector)

``` r
library(rlang)
e <- expr(setcolorder(dt2[dt1, on = .(x), allow.cartesian = TRUE], c("x", "a", "b")))
e2 <- call2("setcolorder", expr(dt2[dt1, on = .(x), allow.cartesian = TRUE]), c("x", "a", "b"))

str(e[[3]])
#>  language c("x", "a", "b")
str(e2[[3]])
#>  chr [1:3] "x" "a" "b"
```

<sup>Created on 2021-03-03 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

Reply:
```R
e <- expr(setcolorder(dt2[dt1, on = .(x), allow.cartesian = TRUE], !!c("x", "a", "b")))
```

😉

Reply:
I thought it was problematic to use `!!` in `expect_equal()` but actually it was only in `expect_snapshot()` and this works nicely!

Reply:
Oh yeah I need to fix that 😞